api/s2n.h

```diff
@@ -368,7 +368,13 @@ extern const uint8_t *s2n_connection_get_ocsp_response(struct s2n_connection *co
 S2N_API
 extern const uint8_t *s2n_connection_get_sct_list(struct s2n_connection *conn, uint32_t *length);
 
-typedef enum { S2N_NOT_BLOCKED = 0, S2N_BLOCKED_ON_READ, S2N_BLOCKED_ON_WRITE, S2N_BLOCKED_ON_APPLICATION_INPUT } s2n_blocked_status;
+typedef enum {
+    S2N_NOT_BLOCKED = 0,
+    S2N_BLOCKED_ON_READ,
+    S2N_BLOCKED_ON_WRITE,
+    S2N_BLOCKED_ON_APPLICATION_INPUT,
+    S2N_BLOCKED_ON_EARLY_DATA,
```

Comment:
Is adding a new enum variant a breaking change?

Reply:
I worried about that, but [we've done it before](https://github.com/aws/s2n-tls/commit/1a6eb27e06c1f2d8dca8a540ab6fe5c729f5360a). Probably at least worth a minor version bump though.

If we can't add new blocked statuses, then I _could_ reuse S2N_BLOCKED_ON_APPLICATION_INPUT. But I worry that isn't as clear and could confuse customers who use both early data and one of the async functions.